dts/bindings/sensor/nxp,fxls8974-common.yaml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: FXLS8974 3-axis accelerometer sensor
+
+include: sensor-device.yaml
+
+properties:
+  reset-gpios:
+    type: phandle-array
+    description: |
+      RST pin
+      This pin defaults to active high when consumed by the sensor.
+      The property value should ensure the flags properly describe
+      the signal that is presented to the driver.
+
+  int1-gpios:
+    type: phandle-array
+    description: |
+      INT1 pin
+      This pin defaults to active low when produced by the sensor.
+      The property value should ensure the flags properly describe
+      the signal that is presented to the driver.
+
+  int2-gpios:
+    type: phandle-array
+    description: |
+      INT2 pin
+      This pin defaults to active low when produced by the sensor.
+      The property value should ensure the flags properly describe
+      the signal that is presented to the driver.
+
+  range:
+    type: int
+    default: 16
+    description: Range in g
+    enum:
+      - 16 # 16g (7.81 mg/LSB)
+      - 8 # 8g (3.91 mg/LSB)
+      - 4 # 4g (1.95 mg/LSB)
+      - 2 # 2g (0.98 mg/LSB)
```

Comment:
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Also need to move the comments on range encodings into the description so they'll be rendered in the documentation.

Reply:
We've added the sample because we wanted to have a way to test out the accelerometer driver with interrupts, similar to the fxos8600 sample. Also there may be more accelerometers from NXP which could run on this driver in the future. Is there any sample which tests accelerometers with interrupts?